specifications/api/swagger.yaml

```diff
@@ -460,6 +465,20 @@ x-transactionOutputs: &transactionOutputs
       address: *addressId
       amount: *transactionAmount
 
+x-transactionWithdrawals: &transactionWithdrawals
+  description: A list of withdrawals from stake addresses.
+  type: array
+  minItems: 0
+  items:
+    type: object
+    additionalProperties: false
+    required:
+      - stake_address
+      - amount
+    properties:
+      stake_address: *stakeAddress
+      amount: *amount
```

Comment:
Here, I've chosen not to represent withdrawals as a key:value map in the API because it would otherwise be a key:value map where all keys are pretty much unknown by the client :no_mouth: ... wallet clients aren't typically aware of what their stake address is, so having something like:

```json
{ "stake_addr1sjck9mdmfyhzvjhydcjllgj9vjvl522w0573ncustrrr2rg7h9azg4cyqd36yyd48t5ut72hgld0fg2x": 42 }
```

Would make it quite hard for them to inspect the content of that value ^^"... Hence, in the API, I am turning the `Map` we have internally into a list of object with a `stake_address` and `amount` properties.

Reply:
yeah, sound decision

Reply:
ok, was thinking how absent withdrawals are going to be represented:  either by 0-length array or by making `withdrawals` in `ApiTransaction` not required...

Reply:
It'll be a 0-length array.